XPCBackupService/Services/BackupUploadService.swift

```diff
@@ -150,36 +189,74 @@ struct BackupUploadService {
 
             self.logger.info("Upload completed with id \(result.id)")
 
-            let stringRemoteParentId = "\(remoteParentId)"
+            if node.syncStatus == .NEEDS_UPDATE {
+                // UPDATE FILE REFERENCE
+                guard let remoteUuid = node.remoteUuid, let remoteId = node.remoteId else {
+                    throw BackupUploadError.MissingRemoteId
+                }
 
-            let encryptedFilename = try encrypt.encrypt(
-                string: filename.deletingPathExtension,
-                password: "\(config.CRYPTO_SECRET2)-\(stringRemoteParentId)",
-                salt: cryptoUtils.hexStringToBytes(config.MAGIC_SALT_HEX),
-                iv: Data(cryptoUtils.hexStringToBytes(config.MAGIC_IV_HEX))
-            )
+                let updatedFile = try await backupNewAPI.replaceFileId(
+                    fileUuid: remoteUuid,
+                    newFileId: result.id,
+                    newSize: result.size
+                )
+
+                self.logger.info("✅ Updated file correctly with identifier \(updatedFile.fileId)")
+
+                node.progress.completedUnitCount = 1
 
-            let createdFile = try await backupAPI.createBackupFile(
-                createFileData: CreateFileData(
-                    fileId: result.id,
-                    type: filename.pathExtension,
-                    bucket: result.bucket,
-                    size: result.size,
-                    folderId: remoteParentId,
-                    name: encryptedFilename.base64EncodedString(),
-                    plainName: filename.deletingPathExtension
+                // Edit date in synced database
+                try self.editSyncedNodeDate(remoteUuid: remoteUuid, date: Date.now)
+
+                if encryptedContentURL != nil {
+                    try FileManager.default.removeItem(at: encryptedContentURL!)
+                }
+
+                return (remoteId, remoteUuid)
+            } else {
+                // CREATE FILE REFERENCE
```

Comment:
Remove this comment if is not helping to understand better the code, I believe is explanatory enough right now